n5010 fpga-ofs-dev-5.10-lts

hundeboll-silicom opened the pull request:
A few reverts and then cherry-picks from n5010/fpga-ofs-dev-5.10-lts.

The most interesting changes are probably:

1.  052e3e8a77ec - build: dkms: auto-generate list of modules to install
2.  4a1e1f6b4157 - Import from mainline v5.10.27

Commit 052e3e8a77ec makes DKMS generate a list of built modules to install. This both eases backports (because we don't need to maintain dkms.conf), and it allows us to conditonally build modules. The latter is important for the upcoming drivers for n6010 (Arrow Creek), which depends on regmap-mmio.

Commit 4a1e1f6b4157 adds regmap-mmio to the backport, as it is needed by n6010 (Arrow Creek), but isn't included in RHEL8x kernels.

trixirt commented:
I am not a fan of reverting code just to make a merge easier.
Why do the modules need a hard coded alias ? The explanation should be in the commit log
n6010 that may need regmap should be in a new feature branch 
look at the  last net fix, the silicom code may need a similar change.
I will look at cherry picking parts out of this branch

hundeboll-silicom commented:
I squashed the reverts with the corresponding commits from the n5010 branch, and added short descriptions of the reasons behind. Also added (repeated) descriptions to the module alias commits. And finally, I removed the regmap-mmio part.

I'll submit the mentioned fix for n5010-hssi to the "upstream" `fpga-ofs-dev-5.10-lts` branch.
